docs: remove OCI consideration from BUILDING.md

cbgbt opened the pull request:
**Issue number:**

Closes #591

**Description of changes:**
Twoliter now handles OCI fetching internally -- there is no need for builders to consider the OCI toolchain.


**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

bcressey commented:
We'll need a similar change over in the kernel kit.

jmt-lab commented on BUILDING.md:
```diff
@@ -30,33 +30,6 @@ To get it, run:
-
-We recommend using `crane` (and `krane`) over `docker` as it has shown better performance in our testing.
-
-#### Docker
```
I can get the above, is there a reason we are removing the docker part entirely, like i don't think we need the daemon.json anymore due to modern docker and our move to krane but docker is still a build requirement that is needed.

cbgbt replied:
@jmt-lab totally true, mind taking another look?

cbgbt committed:
docs: remove OCI consideration from BUILDING.md

Twoliter now handles OCI fetching internally -- there is no need
for builders to consider the OCI toolchain.
